tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -163,6 +163,35 @@ def _create_test_dag_run(self, state='running', extra_dag=False, commit=True):
                 session.add_all(dags)
         return dag_runs
 
+    def _create_test_dag_run_with_queued(self, commit=True):
+        dag_runs = []
+        dags = [DagModel(dag_id="TEST_DAG_ID")]
+        dagrun_model_1 = DagRun(
+            dag_id="TEST_DAG_ID",
+            run_id="TEST_DAG_RUN_ID_1",
+            run_type=DagRunType.MANUAL,
+            execution_date=timezone.parse(self.default_time),
+            start_date=timezone.parse(self.default_time),
+            external_trigger=True,
+            state='running',
+        )
+        dag_runs.append(dagrun_model_1)
+        dagrun_model_2 = DagRun(
+            dag_id="TEST_DAG_ID",
+            run_id="TEST_DAG_RUN_ID_2",
+            run_type=DagRunType.MANUAL,
+            execution_date=timezone.parse(self.default_time_2),
+            start_date=timezone.parse(self.default_time),
+            external_trigger=True,
+            state='queued',
+        )
+        dag_runs.append(dagrun_model_2)
+        if commit:
+            with create_session() as session:
+                session.add_all(dag_runs)
+                session.add_all(dags)
+        return dag_runs
```

Comment:
I don't think we need this duplication, we can use `_create_test_dag_run` instead. I understand you want to limit it to 2 dagruns but that's the essence of the PR, we can filter down to 2. `_create_test_dag_run` takes `state` argument. You can call it twice or more with different states thereby creating more dagruns and then test the added filtering.

Reply:
Hi @ephraimbuddy , If we follow that approach we get these errors:
`E       MySQLdb._exceptions.IntegrityError: (1062, "Duplicate entry 'TEST_DAG_ID-2020-06-11 18:00:00.000000' for key 'dag_run_dag_id_execution_date_key'")`

The problem is that the method _create_test_dag_run uses the self.default_time and self.default_time_2 as execution time statically. So when we call the method twice, we have a violation of PK.
How do you advice to proceed?
Thanks.

Reply:
What do you think about changing the method _create_test_dag_run to avoid that, if it's possible let's do it instead

Reply:
Ok @ephraimbuddy, let's try that. Thanks!

Reply:
Dear @ephraimbuddy, I just made a new commit with the changes. Please let me know if you agree with them. Thanks so much.